### PR TITLE
Makes video.js reload with flash only when the player errors.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Makes video.js reload with flash only when the player errors.
Also independently imports vtt.js and uses the videojs no-vtt to prevent unnecessary cdn loads.

## TODO

If not all TODOs are marked, this PR is considered WIP (work in progress)

- [ ] Have **tests** been written for the new code? If you're fixing a bug, write a regression test (or have a really good reason for not writing one... and I mean **really** good!)
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Issues addressed

List the issues solved or partly solved by the PR

Independently imports vtt.js, uses videojs-novtt.

### CODE REVIEW

File: kalite/distributed/static/js/distributed/video/views.js [hunk 1]
```diff
@@ -68,12 +77,28 @@ var VideoPlayerView = ContentBaseView.extend({
...
             this.player = window.player = _V_(player_id, video_player_options);
+            this.player.ready(function() {
+                videoPlayer = this;
+
+                videoPlayer.on('error', function() { // error event listener
+                    // dispose the old player and its HTML
+                    videoPlayer.dispose();
+
+                    self.flash_only = true;
+                    self.render();
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/video/views.js]: Wouldn't you have to set `this._loaded = false;` as well?

Comment (rtibbles) [inline kalite/distributed/static/js/distributed/video/views.js]: Yes, just fixed..

File: kalite/distributed/static/js/distributed/video/views.js [hunk 2]
```diff
@@ -68,12 +77,29 @@ var VideoPlayerView = ContentBaseView.extend({
                 "playbackRates": [0.5, 1, 1.25, 1.5, 2],
                 "html5": {
                     nativeTextTracks: false
+                },
+                "techOrder": this.flash_only ? ["flash"] : ["html5", "flash"],
+                flash: {
+                    swf: window.sessionModel.get("STATIC_URL") + "distributed/js/distributed/video/video-js.swf"
```
Comment (MCGallaspy) [inline kalite/distributed/static/js/distributed/video/views.js]: The url is also wrong -- remove the leading `distributed`

**Comment by MCGallaspy** [comment]
1. `video-js.swf` is fetched locally (and not from a CDN) if you fix the url. :+1: 
2. VTTs don't work if the `vtt.js` file isn't fetched from a CDN, so somehow it's not the right fix.
3. On Windows XP+FF, the `loadedmetadata` event is never triggered at all -- I started trying to figure out how we could initialize the player without listening for that event, as I suspect we could get the correct flash fallback behavior if we do that.

Do you want to continue with this PR and take over the issue? Or should I? If the latter, input on disentangling player initialization from `loadedmetadata` event would be appreciated.

**Comment by MCGallaspy** [comment]
> loadedmetadata event is never triggered at all

The point being that `initialize_player` is never executed.

**Comment by MCGallaspy** [comment]
BTW, if we can't resolve the issues on XP then we need to document a workaround (like using chrome browser).

**Comment by rtibbles** [comment]
I won't be able to fix this today. I would recommend closing this pr and
opening a new one with the right changes.

**Pull request closed without merging**